src/utilities.py

```diff
@@ -214,3 +206,4 @@ def get_next_color(items):
     for color in color_cycle:
         if color not in used_colors:
             return color
+    return "000000"
```

Comment:
In principle it should never reach this last return value. If all colors in the cycle are full, it should have emptied the color_cycle again.

But I don't mind this return value as a potential safeguard anyway in unforeseen edge cases.